### ISSUE TITLE
Add Advanced Query Builder Select Actions

### ISSUE THREAD

**Issue by edwinmp** [devinit/ddw-analyst-ui#520]
While the JSON can be edited manually, `ACTIONS` are helpers that aid in this editing, reducing on errors. 

The Select `Action` adds the following properties to the QB JSON:

1. `selectAll` - boolean
2. `columns` - object array - if `selectAll` is true, this will be used to determine the column order

**Actions**:

- [x] Select Columns - used to select one or more columns. This action replaces all existing contents of the `columns` property with the selected columns.
- [x] Select ALL - a checkbox to toggle the `selectAll` property
- [x] Clear/Reset - Removes select properties
- [x] Aggregate - helps with aggregating selected columns. Will limit columns to number columns

Lastly, validate all select configs i.e.

- [ ] All columns must have valid IDs and names
- [ ] All columns must have an alias

![image](https://user-images.githubusercontent.com/5672438/118520073-5feca180-b742-11eb-8844-1720d7e790d5.png)

**Comment by edwinmp** [comment]
@ThatcherK  For aggregation, the design should be similar to the filter UI shown below

![image](https://user-images.githubusercontent.com/5672438/122059528-66c80c00-cdf5-11eb-8090-0ab6e1e6061d.png)

Except it'll be just two drop-downs, one for selecting a column & the other for selecting the aggregation. The plus will stay, to add the aggregation.

So the behaviour is as follows:

- The aggregate button should only show if there are selected columns
- On clicking the aggregate button, the aggregate widget above is displayed
- The aggregate widget's select dropdown must only show selected columns i.e. columns that were chosen via the select action
- When the plus button is clicked (after selecting the column & aggregate), append the selected aggregate to the respective column in the JSON config.

Questions?

**Comment by edwinmp** [comment]
Depends on issue #554 before implementing validation ... completed features have been merged into #478

**Comment by edwinmp** [comment]
**New Features**

1. On reviewing error messages, it was brought to my attention that when a SQL statements has an aggregation, then the returned columns must either be aggregated or part of the `groupby` clause.

Therefore, if some of the selected columns are NOT aggregated, the default should be to add them to the `groupby` property.

![image](https://user-images.githubusercontent.com/5672438/123621214-a8a07b80-d813-11eb-8b10-b8ac5ff9b61e.png)


2. Only show the aggregate button when at least one of the selected columns supports aggregation i.e. is numerical

**Comment by edwinmp** [comment]
**Bug**

- [ ] Ordering columns removes aggregation
